lib/features/modeling/cmd/AddLaneHandler.js

```diff
@@ -99,3 +100,17 @@ AddLaneHandler.prototype.preExecute = function(context) {
     height: 120
   }, laneParent);
 };
+
+AddLaneHandler.prototype.postExecute = function(context) {
```

Comment:
Why is that not happening automatically, as the participant is resized during lane add? Maybe we need to change the resizing / moving shapes that are affected by the resize operation?

Reply:
The reason is that we disabled re-layouting when using space tool: https://github.com/bpmn-io/diagram-js/blob/aeb6670c09d8e93f25335148656b1754450e44bf/lib/features/modeling/cmd/SpaceToolHandler.js#L67 (via https://github.com/bpmn-io/diagram-js/pull/364). I will look into that if we can just avoid using space tool so that we don't need to re-layout manually.

Reply:
So alternatively we could unbuild the dependency on the space tool and introduce the following flow:
1) resize participant => grow it in the direction depending on position of the added lane
2) move lanes and lanes' children
3) create the actualn lane.

1 & 2 are basically what the space tool does. The only difference will be that we start to rely on automatic layouting of the resize shape handler which was disabled for space tool.

What do you think?

Reply:
> So alternatively we could unbuild the dependency on the space tool and introduce the following flow.

This is exactly what the space tool does. In my opinion it makes little sense to duplicate it.

I guess your solution is fine for the time being --- as you mentioned we've unbuilt all layouting from the space tool with the last release. As a result we need to compensate for the lack thereof in the places we forgot.